keras/src/callbacks/orbax_checkpoint_test.py

```diff
@@ -293,287 +313,371 @@ def test_initial_value_threshold(self):
             os.path.exists(checkpoint_dir), "Checkpoint directory should exist"
         )
 
+    @parameterized.parameters(
+        {
+            "include_metrics": False,
+            "save_on_background": False,
+        },  # basic
+        {
+            "include_metrics": False,
+            "save_on_background": True,
+        },  # background_save
+        {
+            "include_metrics": True,
+            "save_on_background": False,
+        },  # with_metrics
+        {
+            "include_metrics": True,
+            "save_on_background": True,
+        },  # with_metrics_background
+    )
     @pytest.mark.requires_trainable_backend
-    def test_checkpoint_loading(self):
-        """Test that saved checkpoints can be loaded and weights restored."""
+    def test_checkpoint_loading_comprehensive(
```

Comment:
This test should be doing `load_weights` to actually test the "loading" per its name.